operator/src/network/redis.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::BTreeMap;
+
+use k8s_openapi::{
+    api::{
+        apps::v1::StatefulSetSpec,
+        core::v1::{
+            Container, ContainerPort, PodSpec, PodTemplateSpec, ResourceRequirements,
+            ServicePort, ServiceSpec,
+        },
+    },
+    apimachinery::pkg::{
+        api::resource::Quantity, apis::meta::v1::LabelSelector, apis::meta::v1::ObjectMeta,
+        util::intstr::IntOrString,
+    },
+};
+
+use crate::utils::{managed_labels, selector_labels};
+
+pub const REDIS_APP: &str = "redis";
+
+pub fn service_spec() -> ServiceSpec {
+    ServiceSpec {
+        ports: Some(vec![ServicePort {
+            name: Some("redis-ingress".to_owned()),
+            port: 6379,
+            protocol: Some("TCP".to_owned()),
+            target_port: Some(IntOrString::Int(6379)),
+            ..Default::default()
+        }]),
+        selector: selector_labels(REDIS_APP),
+        type_: Some("ClusterIP".to_owned()),
+        ..Default::default()
+    }
+}
+
+pub fn stateful_set_spec() -> StatefulSetSpec {
+    StatefulSetSpec {
+        replicas: Some(1),
+        selector: LabelSelector {
+            match_labels: selector_labels(REDIS_APP),
+            ..Default::default()
+        },
+        service_name: REDIS_APP.to_owned(),
+        template: PodTemplateSpec {
+            metadata: Some(ObjectMeta {
+                labels: selector_labels(REDIS_APP).map(|mut lbls| {
+                    lbls.append(&mut managed_labels().unwrap());
+                    lbls
+                }),
+                ..Default::default()
+            }),
+            spec: Some(PodSpec {
+                containers: vec![Container {
+                    name: REDIS_APP.to_owned(),
+                    image: Some("redis:latest".to_owned()),
+                    image_pull_policy: Some("Always".to_string()),
```

Comment:
Not a big deal but should we change this to IfNotPresent since it's unlikely to change much?